tests/unittests/Foundation/NSDecimalNumberTests.mm

```diff
@@ -0,0 +1,356 @@
+    EXPECT_EQ(0, result._isNegative);
+
+    NSDecimalSubtract(&result, &right, &left, NSRoundPlain);
+    EXPECT_EQ(2, result._mantissa[0]);
+    EXPECT_EQ(1, result._length);
+    EXPECT_EQ(0, result._exponent);
+    EXPECT_EQ(1, result._isNegative);
+
+    //
+    left = [[[[NSDecimalNumber alloc] initWithInteger:100000] autorelease] decimalValue];
+    right = [[[[NSDecimalNumber alloc] initWithDouble:1e5] autorelease] decimalValue];
+
+    NSDecimalSubtract(&result, &left, &right, NSRoundPlain);
+    EXPECT_EQ(0, result._length);
+    EXPECT_EQ(0, result._isNegative);
+    //
+    left = [[[[NSDecimalNumber alloc] initWithUnsignedLongLong:9223372534876] autorelease] decimalValue];
+    right = [[[[NSDecimalNumber alloc] initWithDouble:0] autorelease] decimalValue];
+
+    NSDecimalSubtract(&result, &left, &right, NSRoundPlain);
+    EXPECT_EQ(NSOrderedSame, NSDecimalCompare(&result, &left));
+
+    //
+    unsigned short mantissa[NSDecimalMaxSize] = { 2048, 25428, 61485, 43339, 53, 0, 0, 0 };
+    result = {};
+    left = [[[[NSDecimalNumber alloc] initWithDouble:-10123456e-6] autorelease] decimalValue];
+    right = [[[[NSDecimalNumber alloc] initWithDouble:-1e3] autorelease] decimalValue];
+
+    NSDecimalSubtract(&result, &left, &right, NSRoundPlain);
+    EXPECT_ARRAY_EQUAL(mantissa, result._mantissa);
+    EXPECT_EQ(5, result._length);
+    EXPECT_EQ(0, result._isNegative);
+    EXPECT_EQ(-18, result._exponent);
+
+    //
+    TEST_SUBTRACT([[[NSDecimalNumber alloc] initWithDouble:10.5] autorelease],
+                  [[[NSDecimalNumber alloc] initWithDouble:10] autorelease],
+                  [[[NSDecimalNumber alloc] initWithDouble:0.5] autorelease]);
+
+    TEST_SUBTRACT([[[NSDecimalNumber alloc] initWithDouble:10.5] autorelease],
+                  [[[NSDecimalNumber alloc] initWithDouble:1000] autorelease],
+                  [[[NSDecimalNumber alloc] initWithDouble:-989.5] autorelease]);
+
+    TEST_SUBTRACT([[[NSDecimalNumber alloc] initWithDouble:-10.5] autorelease],
+                  [[[NSDecimalNumber alloc] initWithDouble:10] autorelease],
+                  [[[NSDecimalNumber alloc] initWithDouble:-20.5] autorelease]);
+
+    TEST_SUBTRACT([[[NSDecimalNumber alloc] initWithDouble:10.5] autorelease],
+                  [[[NSDecimalNumber alloc] initWithDouble:-1e3] autorelease],
+                  [[[NSDecimalNumber alloc] initWithDouble:1010.5] autorelease]);
+
+    NSDecimalNumber* d = [[[NSDecimalNumber alloc] initWithDouble:1012345678912345e-120] autorelease];
+    TEST_SUBTRACT(d, d, [NSDecimalNumber zero]);
+
+    TEST_SUBTRACT([[[NSDecimalNumber alloc] initWithInteger:2] autorelease], [NSDecimalNumber one], [NSDecimalNumber one]);
+
+    TEST_SUBTRACT([[[NSDecimalNumber alloc] initWithInt:(2 * USHRT_MAX)] autorelease],
+                  [[[NSDecimalNumber alloc] initWithInt:USHRT_MAX] autorelease],
+                  [[[NSDecimalNumber alloc] initWithInt:USHRT_MAX] autorelease]);
+}
```

Comment:
I derped - I was checking at the hex level and I didn't notice that there was an exponent of 10^1.

---
In reply to: [81407456](https://github.com/Microsoft/WinObjC/pull/1032#discussion_r81407456) [](ancestors = 81407456,81239938)